integrations/ultralytics/deepsparse/benchmark.py

```diff
@@ -0,0 +1,430 @@
+
+    parser.add_argument(
+        "model_filepath",
+        type=str,
+        help=(
+            "The full filepath of the ONNX model file or SparseZoo stub to the model "
+            "for deepsparse and onnxruntime benchmarks. Path to a .pt loadable PyTorch "
+            "Module for torch benchmarks - the Module can be the top-level object "
+            "loaded or loaded into 'model' in a state dict"
+        ),
+    )
+
+    parser.add_argument(
+        "-e",
+        "--engine",
+        type=str,
+        default=DEEPSPARSE_ENGINE,
+        choices=[DEEPSPARSE_ENGINE, ORT_ENGINE, TORCH_ENGINE],
+        help=(
+            "Inference engine backend to run benchmark on. Choices are 'deepsparse', "
+            "'onnxruntime', and 'torch'. Default is 'deepsparse'"
+        ),
+    )
+
+    parser.add_argument(
+        "--data-path",
+        type=Optional[str],
+        default=None,
+        help=(
+            "Optional filepath to image examples to run the benchmark on. Can be path to "
+            "directory, single image jpg file, or a glob path. All files should be "
+            "in jpg format. If not provided, sample COCO images will be downloaded from "
+            "the SparseZoo"
+        ),
+    )
+
+    parser.add_argument(
+        "--image-shape",
+        type=int,
+        default=(640, 640),
+        nargs="+",
+        help="Image shape to benchmark with, must be two integers. Default is 640 640",
+    )
+
+    parser.add_argument(
+        "-b",
+        "--batch-size",
+        type=int,
+        default=32,
+        help="The batch size to run the benchmark for",
+    )
+    parser.add_argument(
+        "-c",
+        "--num-cores",
+        type=int,
+        default=CORES_PER_SOCKET,
+        help=(
+            "The number of physical cores to run the benchmark on, "
+            "defaults to all physical cores available on the system"
+        ),
+    )
+    parser.add_argument(
+        "-i",
+        "--num-iterations",
+        help="The number of iterations the benchmark will be run for",
+        type=int,
+        default=80,
+    )
+    parser.add_argument(
+        "-w",
+        "--num-warmup-iterations",
+        help=(
+            "The number of warmup iterations that will be executed before the actual"
+            " benchmarking"
+        ),
+        type=int,
+        default=25,
+    )
+    parser.add_argument(
+        "-q",
+        "--quantized-inputs",
+        help=("Set flag to execute benchmark with int8 inputs instead of float32"),
+        action="store_true",
+    )
+    parser.add_argument(
+        "--fp16",
+        help=("Set flag to execute torch benchmark in half precision (fp16)"),
+        action="store_true",
+    )
+    parser.add_argument(
+        "--device",
+        type=_parse_device,
+        default="cpu",
+        help=(
+            "Device to benchmark the model with. Default is cpu. Non cpu benchmarking "
```

Comment:
Can we give an example for the torch use case here? Also, looks like this is the device id going off of the parse device function, can we support cpu and cuda for torch and default to first device? They could still control device outside of this through standard commands and I feel most users will spin this up on single gpu machines

Reply:
sounds good.  will make the torch default `0` if cuda is available and add some documentation